src/main/java/seedu/address/logic/parser/FilterBuyersCommandParser.java

```diff
@@ -28,26 +30,21 @@ public FilterBuyersCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_CHARACTERISTICS, PREFIX_PRICE,
                 PREFIX_TAG);
 
-        if (argMultimap.getValue(PREFIX_PRICE).isPresent()) {
-            try {
-                return new FilterBuyersCommand(new FilterBuyerByPricePredicate(argMultimap
-                        .getValue(PREFIX_PRICE).get()));
-            } catch (Exception e) {
-                throw new ParseException(e.getMessage());
-            }
+        if (areMoreThanOnePrefixesPresent(argMultimap, PREFIX_PRICE, PREFIX_CHARACTERISTICS, PREFIX_SELLER)
+                || areNoPrefixesPresent(argMultimap, PREFIX_PRICE, PREFIX_CHARACTERISTICS, PREFIX_SELLER)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterBuyersCommand.MESSAGE_USAGE));
         }
-        // BUG: Owing to StringUtil.containsWordIgnoreCase, characteristics can only be a single word.
-        // However, the error message does not show up in the dialog box in the GUI.
-        if (argMultimap.getValue(PREFIX_CHARACTERISTICS).isPresent()) {
-            try {
-                return new FilterBuyersCommand(new FilterBuyerByCharacteristicsPredicate(argMultimap
-                    .getValue(PREFIX_CHARACTERISTICS).get()));
-            } catch (Exception e) {
-                throw new ParseException(e.getMessage());
-            }
+
+        AbstractFilterBuyerPredicate predicate = null;
+
+        if (argMultimap.getValue(PREFIX_PRICE).isPresent()) {
+            Price price = ParserUtil.parsePrice(argMultimap.getValue(PREFIX_PRICE).get());
+            predicate = new FilterBuyerByPricePredicate(price);
```

Comment:
Maybe we should add a TODO comment that states what else needs to be added to this method, for easy reference in future

Same for FilterPropsCommandParser